eventnotifier/http/auth.go

```diff
@@ -0,0 +1,90 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package http
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	"github.com/tidwall/gjson"
+)
+
+type BearerAuthTokenProvider interface {
+	AuthToken(ctx context.Context) (string, error)
+}
+
+// quickly detect if underlying type does not implement interface
+var _ BearerAuthTokenProvider = bearerAuthTokenProvider{}
+
+type bearerAuthTokenProvider struct {
+	Logger    *zap.SugaredLogger
+	URL       string
+	Client    *http.Client
+	Headers   map[string]string
+	TokenPath string
+}
+
+func NewBearerAuthTokenProvider(logger *zap.SugaredLogger, client *http.Client, url string, headers map[string]string, tokenPath string) BearerAuthTokenProvider {
+	return bearerAuthTokenProvider{
+		logger,
+		url,
+		client,
+		headers,
+		tokenPath,
+	}
+}
+
+func (b bearerAuthTokenProvider) AuthToken(ctx context.Context) (string, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, b.URL, nil)
+	if err != nil {
+		return "", fmt.Errorf("unable to create http request for URL %s: %w", b.URL, err)
+	}
+
+	for headerKey, headerValue := range b.Headers {
+		req.Header.Add(headerKey, headerValue)
+	}
+
+	b.Logger.Debugw("sending request to get token", "req", fmt.Sprintf("%+v", req))
+
+	res, err := b.Client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("unable to do http request to get token: %w", err)
+	}
+
+	defer func() {
+		if err = res.Body.Close(); err != nil {
+			b.Logger.Warnw("an error occurred while closing body after reading auth token", "error", err)
+		}
+	}()
+
+	if res.StatusCode >= 300 || res.StatusCode < 200 {
+		return "", fmt.Errorf("received response contains unexpected status code %d when retrieving auth", res.StatusCode)
+	}
+
+	tokenBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return "", fmt.Errorf("error when reading token: %w", err)
+	}
+
+	if b.TokenPath == "" {
+		return string(tokenBytes), nil
+	}
+
+	value := gjson.Get(string(tokenBytes), b.TokenPath)
```

Comment:
We're adding a new dependency for this single `Get` function, is it not possible to parse this data otherwise?

Reply:
It is definitely possible to parse the data in a different way, but the were a couple of reasons to use `gjson`.
It allows to enable per-configuration path queries. It also provides flexibility by allowing us to query and extract data directly from JSON structures without needing to pre-define Go structs for every configuration. Finally, it is optimized for speed and efficiency, which helps avoid the overhead associated with marshaling and unmarshaling.